soc/bcm2835/rng.go

```diff
@@ -44,14 +50,18 @@ func getRandomData(b []byte) {
 	RNG.getRandomData(b)
 }
 
-// Init initializes the RNG by discarding 'warmup bytes'
+// Init initializes the RNG by discarding 'warmup bytes'.
 func (hw *Rng) Init() {
 	hw.Lock()
 	defer hw.Unlock()
 
+	hw.status = PeripheralAddress(RNG_STATUS)
+	hw.data = PeripheralAddress(RNG_DATA)
+	hw.ctrl = PeripheralAddress(RNG_CTRL)
+
 	// Discard
-	reg.Write(PeripheralAddress(RNG_BASE+RNG_STATUS), warmupCount)
-	reg.Write(PeripheralAddress(RNG_BASE+RNG_CTRL), RNG_RBGEN)
+	reg.Write(hw.status, warmupCount)
+	reg.Set(hw.ctrl, CTRL_EN)
```

Comment:
`reg.Set` fails to correctly initialize the hardware - the register should be set to `0x1`, not have the first bit set it appears.  With reg.Set, tamago fails to initialize getting stuck in the loop waiting for random bytes to be available.
```suggestion
	reg.Write(hw.ctrl, CTRL_EN)
```